### DISCUSSION TITLE
Page only working horizontal, not vertikal /Portrait Mode

### DISCUSSION THREAD

**Discussion by OnTourWithGerrit** [guysoft/FullPageOS#577] [Q&A]
As I came across your build of FullPageOS,
I installed it on a RPi 3B.
Installation worked flawless but the page was displayed horizontal.
So I followed you guide (rp4, Option 1) and indeed the screen was now vertical/Portrait Mode.
But the page is not loading now. (It is actually made for Portrait Mode and worked before).
If I change the setting back, activating dtoverlay=vc4-fkms-v3d, the page is then displayed, but again in the wrong mode. Horizontal.

Do you have an Idea?

**Comment by guysoft** [comment]
Switching to discussion since its not a bug in the scope I can fix, its rpi OS stuff

**Comment by guysoft** [comment]
I have no idea, my only guess is that acceleration is behaving strangely in portrait. You can try using the software rotation here:
https://github.com/guysoft/FullPageOS/issues/545#issuecomment-2081254684

**Reply by OnTourWithGerrit** [reply]
Thank you for your reply.
I am still troubleshouting.
To add a few things:
When adding the suggested lines from Wiki, it loads into a black screen and a bar blinking in the corner. Thats it.
It is no longer responding. SSH works, but comands like "reboot" don't.
If i remove them and add the Original content into the config.txt, it loads very fast, the Webpage is showing all animations etc. (Dakboard).
Your suggestion from -545- unfortunately down't show any effect.
I'll keep you updated.

**Reply by guysoft** [reply]
Not really sure how to fix it at this point. I am out of tricks. If you manage to find a way would be happy if you update!

**Reply by OnTourWithGerrit** [reply]
Hi,

a quick update.
So far i found two solutions, not the best ones but working for me.

1: SSH into RPi.
Execute: DISPLAY=:0 xrandr --output HDMI-1 --rotate left
Change last word into normal, right or inverted to rotate in different ways.
Pro: Working from the moment you execute it.
Con: Bad performace (Animations and overall display is sluggish,)
Need to be done after every reboot.

2: I added a CSS Rule to the Page, which need to be displayed

/*Rotate right */
#screen-container-no-edit{
    margin:0;
    transform-origin:bottom left;
    transform: rotate(90deg);
    position:absolute;
    top: -100vw;
    left: 0;
    height:100vw;
    width:100vh;
    overflow:visible;
}

Pro: Working always also after reboot. 
Animations /performace still the same (good performace) like without any changes.
Con: Bootscreen / Text during boot still in "wrong" rotation" But thats it.

**Reply by guysoft** [reply]
Its worth adding that i the wiki:
https://github.com/guysoft/FullPageOS/wiki
